### ISSUE TITLE
How to Install QL-800 on Ubuntu

### ISSUE THREAD

**Issue by shtoiven** [philpem/printer-driver-ptouch#8]
Hi, thank you for your efforts to make this available to Ubuntu, however, I am wondering how to install the P-touch Editor Tool for the Brother QL-800 printer? 

I cannot see a step by step guide to installing the p-touch software on Terminal

I had used P-touch Editor Tool 5.4 on Windows.

If the " install-sh " file is the instructions to install the software, I give up.... Certain instructions are Not beginner friendly, such as 
"set DOITPROG to echo to test this script "  ?????

Can you please provide the step-by-step guide here please :)

Thank you in advance :) 

I have installed other software on ubuntu with 2 or 3 coding lines (seems to source the software from cyberspace).... is this possible for P-touch ?

**Comment by philpem** [comment]
If you're using Ubuntu, it's much easier to install than that.

Use the Software Manager to install the `printer-driver-ptouch` package.

If that doesn't work, open a terminal and type this command:

`sudo apt install printer-driver-ptouch`.

You shouldn't need to build from source for a QL-800, I think that's been supported for some time.

**Comment by shtoiven** [comment]
Thanks, although, how do i get the ptouch program to work?  there should be something called ptouch 5.4

**Comment by shtoiven** [comment]
prior to my msg above, i ran "sudo apt install printer-driver-ptouch" which worked. but i still have no ' editor tool ' program to open ?

**Comment by philpem** [comment]
The P-Touch Editor is created by Brother and there is no Linux version that I know of. This driver (printer-driver-ptouch / ptouch-driver) was developed completely from scratch without access to Brother's code, or assistance from them beyond the programming documentation.

You should have "gLabels" available in the Software Manager - this is similar to the P-Touch Editor but works with other brands of label printer too.

You may also be able to create labels using Openoffice, Libreoffice or Inkscape - which you use is really up to you. The driver doesn't mind, it just takes the images from your software and formats them for the printer.

**Comment by philpem** [comment]
> prior to my msg above, i ran "sudo apt install printer-driver-ptouch" which worked. but i still have no ' editor tool ' program to open ?

printer-driver-ptouch is only a printer driver, not an application.

**Comment by shtoiven** [comment]
Thanks. I am playing with Inkscape - is there a way to set the cut line so that the printer knows to cut at certain intervals?
Also, when I test the printer - it continually says "waiting for the printer to become available"

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by shtoiven** [comment]
[
![Screenshot from 2021-03-07 11-02-08](https://user-images.githubusercontent.com/80145113/110225365-ffb4d480-7f34-11eb-8afe-64e9e39bc26b.png)
](url)

**Comment by philpem** [comment]
Why did you attach the screenshot? Firstly, please stop trying things at random without understanding what they do...!

What version of Ubuntu are you using?

For the printer to work, you need to have the label type (pre-cut or continuous) and label dimensions set correctly. You also need to send an image of the correct size to the driver -- that means the paper size in Inkscape must be the same size as the label.

You may find the V1.5.1 driver more reliable - there were a lot of bug fixes.

Given the long list of commands you were seemingly trying to run at random in the screenshot, I get the impression you have never installed a package from source code before. If you want to do this with the P-touch driver, you will need to:

* remove the printer-driver-ptouch package first (`apt remove printer-driver-ptouch`).
* run `./autogen.sh` to build the configure script
* follow the instructions in the `INSTALL` file (briefly: `./configure && make && sudo make install`).

**Comment by shtoiven** [comment]
Will try those. Also, have you seen the following driver from Brother before? I tried installing it beforehand.

https://support.brother.com/g/b/downloadend.aspx?c=au&lang=en&prod=lpql800eas&os=130&dlid=dlfp100337_000&flang=178&type3=10309&dlang=true

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by shtoiven** [comment]
I'm using 20.04 .. The attached screenshot shows the error i confronted when running the command " ./autogen.sh "

Regarding "send an image of the correct size to the driver" - do you mean to go to 'document properties' on Inkscape - and then select the paper size as B8 (seeing that i am using 62mm labels) - Wouldn't the issue with the B8 size label be that the length of a label would be constrained to 88mm, even though the printer can do any length?...
Is there still a standard 62mm width label option when using Ubuntu?

Also the command " ./configure " results in an error " configure: error: cannot find sources (rastertoptch.c) in . or .. "

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by shtoiven** [comment]
I have tried installing 1.5.1 too. following error

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by philpem** [comment]
> I'm using 20.04 .. The attached screenshot shows the error i confronted when running the command " ./autogen.sh "

There is no attached screenshot which shows the output of autogen.sh. The screenshot you sent earlier shows you running a bunch of random commands.

You need to read the documentation and understand what you're doing. Running random commands like a monkey at a typewriter will not help you, it will only make a bigger mess which is harder to fix.


> Regarding "send an image of the correct size to the driver" - do you mean to go to 'document properties' on Inkscape - and then select the paper size as B8 (seeing that i am using 62mm labels) - Wouldn't the issue with the B8 size label be that the length of a label would be constrained to 88mm, even though the printer can do any length?... Is there still a standard 62mm width label option when using Ubuntu?

You need to set a custom paper size in the Page Options, and also in the printer settings. The width should be the same as the label width, printed on the side of the label cartridge. The height should be either the length of the label if you're using pre-cut ones, or the length of the label you want if you're using continuous tape.

I've never used Inkscape with this driver, only gLabels. gLabels has predefined label sizes for the Brother labels and will handle the print settings for you.

Make sure the label type is correct in Print Settings too -- either Die Cut or Continuous.

You should try to get things working in gLabels first, as this is known to work with the driver and has predefined settings for the Brother label stock.


> Also the command " ./configure " results in an error " configure: error: cannot find sources (rastertoptch.c) in . or .. " Sent with [ProtonMail](https://protonmail.com) Secure Email. ‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

Can you please send your replies on Github instead of by email? Replying to the emails (and replying several times) is making the ticket hard to follow.


> I have tried installing 1.5.1 too. following error Sent with ProtonMail

There is no error text included in this last message.

**Comment by philpem** [comment]
> Will try those. Also, have you seen the following driver from Brother before? I tried installing it beforehand. https://support.brother.com/g/b/downloadend.aspx?c=au&lang=en&prod=lpql800eas&os=130&dlid=dlfp100337_000&flang=178&type3=10309&dlang=true Sent with [ProtonMail](https://protonmail.com) Secure Email. ‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

You need to uninstall the Brother driver before using printer-driver-ptouch. You should not have two drivers installed for the same printer. Having the Brother and open-source drivers installed is not supported.
